### PR TITLE
include <dataset> ID in siphon.Dataset

### PR DESCRIPTION
**Pull request by juozasg**
The 'ID' attribute from THREDDS catalog is useful for some purposes and should be parsed

### CODE REVIEW

**Comment by lesserwhirls** [comment]
Good catch! My only comment would be that since `ID` is an optional attribute on the `dataset` element in THREDDS client catalog [spec](https://www.unidata.ucar.edu/software/thredds/current/tds/catalog/InvCatalogSpec.html#dataset), I think we should leave it unset if it does not exist, rather than using the value of  the `name` attribute.

File: siphon/catalog.py [hunk 1]
```diff
@@ -369,6 +369,10 @@ def __init__(self, element_node, catalog_url=''):
 
         """
         self.name = element_node.attrib['name']
+        if 'ID' in element_node.attrib:
+            self.id = element_node.attrib['ID']
+        else:
+            self.id = self.name
```
Comment (jrleeman) [inline siphon/catalog.py]: I tend to agree with @lesserwhirls here. I think setting to `None` is probably the best default behavior.

**Comment by jrleeman** [comment]
@juozasg - Would you mind signing the CLA? We can fixup everything else.

**Comment by jrleeman** [comment]
Closed in favor of #251

**Pull request closed without merging**